### ISSUE TITLE
Bug: `llama-server` web UI resets the text selection during inference on every token update

### ISSUE THREAD

**Issue by mashdragon** [ggerganov/llama.cpp#9608]
### What happened?

When using `llama-server`, the output in the UI can't be easily selected or copied until after text generation stops. This may be because the script replaces all the DOM nodes of the current generation when every new token is output.

The existing text content ideally shouldn't be replaced during generation so we can copy the text as it continues to produce output.

### Name and Version

version: 3755 (822b6322)
built with cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0 for x86_64-linux-gnu

**Comment by mashdragon** [comment]
This is still an issue which impacts usability.

**Comment by ngxson** [comment]
this should now be resolved, we have a dedicated copy button on the new UI

**Comment by mashdragon** [comment]
No, sadly the same issue still is there. During generation, try selecting some text. The text nodes are immediately overwritten, causing your selection to disappear.

**Comment by ngxson** [comment]
Hmm ok, there is not much we can do about that because the text is re-render by `markdown-it` (to support markdown rendering).

**Comment by slaren** [comment]
Let's keep it open in case anyone can figure a way to fix it.

**Comment by mashdragon** [comment]
Thank you @slaren. Since many other Markdown renderers (on ChatGPT and others) support this, I'm sure it's possible. If I have any free time in the near future, I can try to swap out markdown-it for a different solution.

**Comment by ngxson** [comment]
IMO the incremental DOM is quite bad for performance, since it have to parse the html string into DOM, do diff, then modify the real DOM.

Another option could be moving to [remark](https://github.com/medfreeman/remark-vue) which transform markdown directly to DOM or vue component. This package the quite old, but I'm sure there are modern equivalent to it (I will have a look later)

The implementation on chatgpt is virtual DOM, though it's transparent to the developer because the library that they use, `micromark`, already handle it under the hood (but it's quite complicate to plug into vue)

**Comment by ngxson** [comment]
Fixed in https://github.com/ggerganov/llama.cpp/pull/11688
